phpBB/phpbb/install/module/update_database/task/update_extensions.php

```diff
@@ -0,0 +1,244 @@
+			->ignoreUnreadableDirs()
+			->depth('< 3')
+			->files()
+			->name('composer.json');
+
+		// Make sure asset version exists in config. Otherwise we might try to
+		// insert the assets_version setting into the database and cause a
+		// duplicate entry error.
+		if (!isset($this->config['assets_version']))
+		{
+			$this->config['assets_version'] = 0;
+		}
+
+		parent::__construct(true);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function run()
+	{
+		$this->user->session_begin();
+		$this->user->setup(array('common', 'acp/common', 'cli'));
+
+		$update_info = $this->install_config->get('update_info_unprocessed', []);
+		$version_from = !empty($update_info) ? $update_info['version']['from'] : $this->config['version_update_from'];
+
+		if (!empty($version_from))
+		{
+			$update_extensions = $this->iohandler->get_input('update-extensions', []);
+
+			// Create list of default extensions that need to be enabled in update
+			$default_update_extensions = [];
+			foreach (self::$default_extensions_update as $version => $extensions)
+			{
+				if ($this->update_helper->phpbb_version_compare($version_from, $version, '<='))
+				{
+					$default_update_extensions = array_merge($default_update_extensions, $extensions);
+				}
+			}
+
+			$available_extensions = $this->extension_manager->all_available();
+
+			// Update available extensions
+			foreach ($available_extensions as $ext_name => $ext_path)
+			{
+				// Update extensions if:
+				//	1) Extension is currently enabled
+				//	2) Extension was implicitly defined as needing an update
+				//	3) Extension was newly added as default phpBB extension in
+				//		this update and should be enabled by default.
+				if ($this->extension_manager->is_enabled($ext_name) ||
+					in_array($ext_name, $update_extensions) ||
+					in_array($ext_name, $default_update_extensions)
+				)
+				{
+					try
+					{
+						$extension_enabled = $this->extension_manager->is_enabled($ext_name);
+						if ($extension_enabled)
+						{
+							$this->extension_manager->disable($ext_name);
+						}
+						$this->extension_manager->enable($ext_name);
+						$extensions = $this->get_extensions();
+
+						if (isset($extensions[$ext_name]) && $extensions[$ext_name]['ext_active'])
+						{
+							// Create log
+							$this->log->add('admin', ANONYMOUS, '', 'LOG_EXT_ENABLE', time(), array($ext_name));
+							$this->iohandler->add_success_message(array('CLI_EXTENSION_ENABLE_SUCCESS', $ext_name));
+						} else
+						{
+							$this->iohandler->add_log_message('CLI_EXTENSION_ENABLE_FAILURE', array($ext_name));
+						}
+
+						// Disable extensions if it was disabled by the admin before
+						if (!$extension_enabled && !in_array($ext_name, $default_update_extensions))
+						{
+							$this->extension_manager->disable($ext_name);
+						}
+					}
+					catch (\Exception $e)
+					{
+						// Add fail log and continue
+						$this->iohandler->add_log_message('CLI_EXTENSION_ENABLE_FAILURE', array($ext_name));
```

Comment:
Not sure if we should use another log message here like "Updating extension %s failed" but I leave this up to you, just mentioning it.